### PR TITLE
Update changelog with notes for v1.4.0

### PR DESCRIPTION
**Pull request by MarcosDY**
Update changelog to v1.4.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,32 @@
 # Changelog
 
+## [1.4.0] - 2022-08-08
+
+### Added 
+- Support K8S worklod attestor using windows containers (#3191)
```
Comment (azdagron) [inline CHANGELOG.md:6]:
```suggestion
- Support for Windows workload attestation on Kubernetes (#3191)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+- Support custom key types used for Workload SVIDs (#3237)
```
Comment (azdagron) [inline CHANGELOG.md:7]:
```suggestion
- Support for using RSA keys with Workload X509-SVIDs (#3237)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,32 @@
...
+- Support anonymous authentication on k8s WorkloadAttestor (#3273)
```
Comment (azdagron) [inline CHANGELOG.md:8]:
```suggestion
- Support for anonymous authentication with the Kubelet when performing workload attestation on Kubernetes (#3273)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,32 @@
...
+
+### Deprecated
+- The Node Resolver plugin type (#3272)
+
+### Fixed
+- Ensures the can_reattest flag is propogated when the attested node record is updated (#3292)
```
Comment (azdagron) [inline CHANGELOG.md:14]:
```suggestion
- Persistence of the can_reattest flag during agent SVID renewal (#3292)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,32 @@
...
+- A regression in Server API authentication that prevents Agents to be notified when it was evicted (#3269)
```
Comment (azdagron) [inline CHANGELOG.md:15]:
```suggestion
- A regression in behavior preventing an agent from re-attesting when it has been evicted (#3269)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,32 @@
...
+
+### Changed
+- The server's Azure node attestor subsumed all the functionality of the node resolver, which has been deprecated (#3272)
+- Fail when prividing unknown configurations to Docker Workload Attestor (#3243)
+- Add special case on K8S Workload attestor when CRI-O does not provide POD ID on cgroup (#3183)
+- Consolidate agent storage information into a single JSON file (#3201)
+- Agent and server data directories on Windows no longer inherited permission from parent directory (#3227)
```
Comment (azdagron) [inline CHANGELOG.md:22]:
```suggestion
- Agent and server data directories on Windows no longer inherit permissions from parent directory (#3227)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,32 @@
...
+
+### Fixed
+- Ensures the can_reattest flag is propogated when the attested node record is updated (#3292)
+- A regression in Server API authentication that prevents Agents to be notified when it was evicted (#3269)
+
+### Changed
+- The server's Azure node attestor subsumed all the functionality of the node resolver, which has been deprecated (#3272)
+- Fail when prividing unknown configurations to Docker Workload Attestor (#3243)
+- Add special case on K8S Workload attestor when CRI-O does not provide POD ID on cgroup (#3183)
+- Consolidate agent storage information into a single JSON file (#3201)
```
Comment (azdagron) [inline CHANGELOG.md:21]:
```suggestion
- Agent data storage has been consolidated to a single JSON file (#3201)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,32 @@
...
+- The Node Resolver plugin type (#3272)
+
+### Fixed
+- Ensures the can_reattest flag is propogated when the attested node record is updated (#3292)
+- A regression in Server API authentication that prevents Agents to be notified when it was evicted (#3269)
+
+### Changed
+- The server's Azure node attestor subsumed all the functionality of the node resolver, which has been deprecated (#3272)
+- Fail when prividing unknown configurations to Docker Workload Attestor (#3243)
+- Add special case on K8S Workload attestor when CRI-O does not provide POD ID on cgroup (#3183)
```
Comment (azdagron) [inline CHANGELOG.md:20]:
```suggestion
- Improved CRI-O support with Kubernetes workload attestation (#3183)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,32 @@
...
+### Deprecated
+- The Node Resolver plugin type (#3272)
+
+### Fixed
+- Ensures the can_reattest flag is propogated when the attested node record is updated (#3292)
+- A regression in Server API authentication that prevents Agents to be notified when it was evicted (#3269)
+
+### Changed
+- The server's Azure node attestor subsumed all the functionality of the node resolver, which has been deprecated (#3272)
+- Fail when prividing unknown configurations to Docker Workload Attestor (#3243)
```
Comment (azdagron) [inline CHANGELOG.md:19]:
```suggestion
- The Docker Workload Attestor now fails when configured with unknown options (#3243)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,32 @@
...
+
+### Deprecated
+- The Node Resolver plugin type (#3272)
+
+### Fixed
+- Ensures the can_reattest flag is propogated when the attested node record is updated (#3292)
+- A regression in Server API authentication that prevents Agents to be notified when it was evicted (#3269)
+
+### Changed
+- The server's Azure node attestor subsumed all the functionality of the node resolver, which has been deprecated (#3272)
```
Comment (azdagron) [inline CHANGELOG.md:18]:
```suggestion
- The Azure Node Attestor to optionally provide selectors (#3272)
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,32 @@
...
+- Fail when prividing unknown configurations to Docker Workload Attestor (#3243)
+- Add special case on K8S Workload attestor when CRI-O does not provide POD ID on cgroup (#3183)
+- Consolidate agent storage information into a single JSON file (#3201)
+- Agent and server data directories on Windows no longer inherited permission from parent directory (#3227)
+- Endpoints exposed using named pipes explicitly deny access to remote callers (#3236)
+- Small documentation improvements (#3264)
+
+### Removed
+- Remove webhood mode from k8s-workload-registrar (#3235)
```
Comment (azdagron) [inline CHANGELOG.md:27]:
```suggestion
- The deprecated webhook mode from the k8s-workload-registrar (#3235)
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,32 @@
...
+- The configmap leader election lock is not longer supported (#3241)
```
Comment (azdagron) [inline CHANGELOG.md:28]:
```suggestion
- Support for the configmap leader election lock type from the k8s-workload-registrar (#3241)
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,32 @@
...
+- The Node Resolver plugin type (#3272)
+
+### Fixed
+- Persistence of the can_reattest flag during agent SVID renewal (#3292)
+- A regression in behavior preventing an agent from re-attesting when it has been evicted (#3269)
+
+### Changed
+- The Azure Node Attestor to optionally provide selectors (#3272)
+- The Docker Workload Attestor now fails when configured with unknown options (#3243)
+- Improved CRI-O support with Kubernetes workload attestation (#3183)
```
Comment (rturner3) [inline CHANGELOG.md:20]:
```suggestion
- Improved CRI-O support with Kubernetes workload attestation (#3242)
```

We usually link the PRs here, not the issues

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,32 @@
...
+- Agent data storage has been consolidated to a single JSON file (#3201)
```
Comment (rturner3) [inline CHANGELOG.md:21]:
```suggestion
- Agent data stored on disk has been consolidated to a single JSON file (#3201)
```
nit: clarifies that this is only affecting the existing files managed on-disk and not data cached in-memory

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,32 @@
 # Changelog
 
+## [1.4.0] - 2022-08-08
+
+### Added 
+- Support for Windows workload attestation on Kubernetes (#3191)
+- Support for using RSA keys with Workload X509-SVIDs (#3237)
+- Support for anonymous authentication with the Kubelet when performing workload attestation on Kubernetes (#3273)
```
Comment (rturner3) [inline CHANGELOG.md:8]:
```suggestion
- Support for anonymous authentication to the Kubelet secure port when performing workload attestation on Kubernetes (#3273)
```

**Review by rturner3 — APPROVED** [review]

**Review by azdagron — APPROVED** [review]

**Pull request merged**